saleor/plugins/avatax/plugin.py

```diff
@@ -595,9 +595,11 @@ def _calculate_order_shipping(
                 return TaxedMoney(net=net, gross=gross)
 
         # Ignore typing checks because it is checked in _validate_order
-        channel_listing = order.shipping_method.channel_listings.filter(  # type: ignore
-            channel_id=order.channel_id
-        ).first()
+        channel_listing = None
+        if shipping_method := order.shipping_method:
+            channel_listing = shipping_method.channel_listings.filter(
+                channel_id=order.channel_id
+            ).first()
```

Comment:
Maybe we should use `base_order_shipping` from [base_calculations.py](https://github.com/mirumee/saleor/blob/2bec9f97f90e7f6587daaf9df01691e718a41cda/saleor/order/base_calculations.py#L14)

Reply:
I do not think so. It's a plugins method, we usually return `previous_value` when something is missing.